src/utilities/fastify/plugins/fastifyActiveEndpoint.ts

```diff
@@ -0,0 +1,19 @@
+import type { FastifyInstance } from 'fastify';
+import fastifyPlugin from 'fastify-plugin';
+
+import { InternetEndpointManager } from '../../awala/InternetEndpointManager.js';
+
+async function fastifyActiveEndpoint(fastify: FastifyInstance) {
+  const endpointManager = await InternetEndpointManager.init(fastify.mongoose);
+  const activeEndpoint = await endpointManager.getActiveEndpoint();
+
+  fastify.decorate('getActiveEndpoint', async () => {
+    await activeEndpoint.makeInitialSessionKeyIfMissing();
+    return activeEndpoint;
+  });
+}
+
+const fastifyActiveEndpointPlugin = fastifyPlugin(fastifyActiveEndpoint, {
```

Comment:
You still need to specify that we need `fastify.mongoose` somewhere -- we can't lose that information.

See: https://github.com/fastify/fastify-plugin#dependencies. Set either `dependencies` or `decorators`, but not both.